bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -1061,14 +1061,15 @@ private Map<Integer, BookieId> getReplacementBookies(
             }
         }
         return getReplacementBookiesByIndexes(
-                lh, ensemble, bookieIndexesToRereplicate, Optional.of(bookiesToRereplicate));
+                lh, ensemble, bookieIndexesToRereplicate, Optional.of(bookiesToRereplicate), false);
```

Comment:
This method's entrance is `recoverBookieData`. The user specifies a bookieSrc and then replaces the data of this bookie with another node. In this scenario, the ledger of the bookie is not checked by ledgerChecker to determine if any data is lost. Instead, the data of the bookie is directly replicated to other bookies. If we set downgradeToSelf to `true`, the data may be copied to bookieSrc itself instead of throwing an exception. I am not sure if this is a good behavior for the user. What do you think?

Reply:
Ok, it makes sense. Make the downgrade logic is the default behavior.